internal/graph/graph.go

```diff
@@ -94,7 +94,7 @@ func (g *Graph) InsertConstructor(ctor interface{}) error {
 		objTypes[i] = ctype.Out(i)
 	}
 
-	if err := g.ValidateReturnTypes(ctype); err != nil {
+	if err := g.ValidateReturnTypes(ctype, false); err != nil {
```

Comment:
Just trying to reuse `ValidateReturnTypes` method. checkCachedObjects=false ignores cached objects for ctor provide and builds the graph to resolve later.